### PR TITLE
Refactored appheadernavigation.main to use hooks

### PR DESCRIPTION
**Pull request by dusanradovanovic**
Description:

This PR adds:
- Refactored `appheadernavigation.main` to be functional component.
- Adds `useFetchCategories` custom hook to separate fetching logic from presentation.

Linting:

- [x] No linting errors

Tests:

- [x] E2E tests (npm test run with `e2e`)
- [x] Manual tests
- [x] Accessibility tests (no new react-axe errors in console)

Documentation:

- [ ] Requires documentation updates
- [ ] Requires Storybook component updates

### CODE REVIEW

File: src/hooks/use-fetch-categories.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+import { useState, useEffect } from 'react';
```
Comment (shaunmaharaj) [inline src/hooks/use-fetch-categories.ts:22]: This is beautiful! so clean! 😭

File: package.json [hunk 2]
```diff
@@ -92,6 +92,7 @@
     "styled-components": "^4.4.1",
     "terser-webpack-plugin": "1.3.0",
     "url-loader": "2.0.0",
+    "use-onclickoutside": "^0.3.1",
```
Comment (shaunmaharaj) [inline package.json:95]: I don't see useOnClickOutside used anywhere. Is this for future-proofing?

**Comment by shaunmaharaj** [comment]
Need to add hover state to sub-categories:
image

Compare against live for reference:
image

**Comment by dusanradovanovic** [comment]
I've added hover state and darkened it a bit. It was barely visible to me 🧐

image

**Pull request merged**